File: go/2643.一最多的行.go
/*
 * @lc app=leetcode.cn id=2643 lang=golang
 *
 * [2643] 一最多的行
 */

package main

// @lc code=start
func rowAndMaximumOnes(mat [][]int) []int {
	if len(mat) == 0 {
		return []int{0, 0}
	}

	max := 0
	maxIndex := 0
	for i, row := range mat {
		count := 0
		for _, val := range row {
			if val == 1 {
				count++
			}
		}
		if count > max {
			max = count
			maxIndex = i
		}
	}
	return []int{maxIndex, max}
}

// @lc code=end

File: typescript/2643.一最多的行.ts
/*
 * @lc app=leetcode.cn id=2643 lang=typescript
 *
 * [2643] 一最多的行
 */

// @lc code=start
function rowAndMaximumOnes(mat: number[][]): number[] {
  if (mat.length === 0) {
    return [0, 0];
  }
  
  let column = 0
  let number = 0
  for (let i = 0; i < mat.length; i++) {
      const currNumber = mat[i].reduce((pre, cur) => pre + cur, 0)
      if (currNumber > number) {
          column = i
          number = currNumber
      }
  }
  return [column, number]
};
// @lc code=end

export { rowAndMaximumOnes };

